recipes/newrelic/infrastructure/ohi/mongodb/debian.yml

```diff
@@ -62,24 +59,344 @@ preInstall:
-      exit 132
-    fi
-    exit 1
+    # TODO: Fix this. If Auth or SSL enable, this required discovery step fails
+    # MONGODB_EXISTS=$(mongo --eval 'db.runCommand({ connectionStatus: 1 })' | grep -e '\"ok\" : 1' | wc -l)
+    # if [ $MONGODB_EXISTS -eq 0 ]; then
+    #   exit 132
+    # fi
+    exit 0
 
 install:
   version: "3"
   silent: true
   tasks:
     default:
       cmds:
-        - task: setup
+        - task: assert_pre_req
+        - task: mongo_pre_req
+        - task: assert_input
+          vars:
+            MAX_RETRIES: 3
+        - task: restart
+        - task: cleanup
+
+    assert_pre_req:
+      cmds:
+        - |
+          IS_INFRA_INSTALLED=$(sudo ps aux | grep newrelic-infra-service | grep -v grep | wc -l)
+          if [ $IS_INFRA_INSTALLED -eq 0 ] ; then
+            echo "The infrastructure agent is required to install this integration, we recommend going through our guided install path for this pre-requisite which can be found here:  https://docs.newrelic.com/docs/full-stack-observability/observe-everything/get-started/new-relic-guided-install-overview" >&2
+            exit 1
+          fi
+        - |
+          # Map of tool names to the associated error code
+          REQUIRED_TOOLS_AND_ERROR_CODES="date:41 sha256sum:42 base64:43 head:44"
+
+          for tuple in $REQUIRED_TOOLS_AND_ERROR_CODES; do
+            tool=$(echo ${tuple} |cut -d':' -f1)
+            code=$(echo ${tuple} |cut -d':' -f2)
+
+            IS_TOOL_INSTALLED=$(which ${tool} | wc -l)
+            if [ "$IS_TOOL_INSTALLED" -eq 0 ]
+            then
+              echo -e "{{.RED}}This installation requires '${tool}' to be installed.{{.NOCOLOR}}" >> /dev/stderr
+              exit ${code}
+            fi
+          done
+        - | 
+          IS_MONGO_SHELL_INSTALLED=$(which mongo | wc -l)
+          if [ $IS_MONGO_SHELL_INSTALLED -eq 0 ]; then
+            echo "Mongo shell is required to run the newrelic install" # TODO: Prompt to install if not present?
+            exit 46
+          fi
+
+    mongo_pre_req:
+      cmds:
+        - |
+          if [ -f /tmp/mongo-role.js ]; then
+            rm -f /tmp/mongo-role.js 
+          fi
+          tee -a /tmp/mongo-role.js > /dev/null <<-"EOT"
+            use admin
+            db.createRole({ role: "listCollections", privileges: [{ resource: { db: "", collection: "" }, actions: [ "listCollections" ] }], roles: [] })
+          EOT
+          if [ -f /tmp/mongo-user.js ]; then
+            rm -f /tmp/mongo-user.js 
+          fi
+          tee -a /tmp/mongo-user.js > /dev/null <<-"EOT"
+            use admin
+            db.createUser({ user: "MONGO_MONITORING_USERNAME", pwd: "MONGO_MONITORING_PASSWORD", roles: [ "clusterMonitor", "listCollections" ] })
+          EOT
+      
+    assert_input:
+      cmds:
+        - |
+          NEW_RELIC_ASSUME_YES="{{.NEW_RELIC_ASSUME_YES}}" # TODO: Implement NEW_RELIC_ASSUME_YES
+          NR_CLI_DB_HOSTNAME="{{.NR_CLI_DB_HOSTNAME}}"
```

Comment:
For all the env vars, we should allow them to be passed in explicitly when running the CLI.
Also we should default appropriately when not specified. NR_CLI_DB_HOSTNAME should be localhost as we dont support installing on remote host with this installation.